include/aspect/mesh_deformation/interface.h

```diff
@@ -538,7 +538,7 @@ namespace aspect
     bool
     MeshDeformationHandler<dim>::has_matching_mesh_deformation_object () const
     {
-      for (typename std::map<types::boundary_id, std::vector<std::unique_ptr<Interface<dim>>>>::iterator boundary_id
+      for (typename std::map<types::boundary_id, std::vector<std::unique_ptr<Interface<dim>>>>::const_iterator boundary_id
```

Comment:
Can you use a ranged for loop with const auto & here? Also, I would call the variable something else: it is not a boundary id but an iterator.